src/osd/OSD.cc

```diff
@@ -10787,18 +10788,19 @@ void OSD::ShardedOpWQ::_enqueue_front(OpSchedulerItem&& item)
 namespace ceph { 
 namespace osd_cmds { 
 
-int heap(CephContext& cct, const cmdmap_t& cmdmap, Formatter& f,
-	 std::ostream& os)
+std::tuple<int, std::stringstream, bufferlist>
+heap(CephContext& cct, const cmdmap_t& cmdmap, Formatter& f)
 {
+  std::stringstream os;
   if (!ceph_using_tcmalloc()) {
         os << "could not issue heap profiler command -- not using tcmalloc!";
-        return -EOPNOTSUPP;
+	return std::make_tuple(-EOPNOTSUPP, std::move(os), bufferlist());
```

Comment:
```suggestion
	return {-EOPNOTSUPP, std::move(os), bufferlist()};
```

also could you take the opportunity to fix the indent?